rust/connlib/snownet/src/allocation.rs

```diff
@@ -1470,6 +1471,30 @@ mod tests {
         )
     }
 
+    #[test]
+    fn allocation_refresh_timeout_is_reset_after_refresh_request_is_sent() {
+        let mut allocation = Allocation::for_test(Instant::now());
+
+        let allocate = allocation.next_message().unwrap();
+        allocation.handle_test_input(
+            &allocate_response(&allocate, &[RELAY_ADDR_IP4, RELAY_ADDR_IP6]),
+            Instant::now(),
+        );
+
+        let next_timeout = allocation.poll_timeout().unwrap();
+        assert_eq!(next_timeout, allocation.refresh_allocation_at().unwrap());
+
+        allocation.handle_timeout(allocation.poll_timeout().unwrap());
+
+        assert!(allocation.refresh_allocation_at().is_none());
+        assert!(allocation.poll_timeout().unwrap() > next_timeout);
+
+        assert!(allocation
```

Comment:
This is also testing internals and I don't really see the benefit of this assertion. Better check the return value of `poll_transmit`.